app/src/main/java/com/gdsctsec/smartt/ui/weekday/WeekdayActivity.kt

```diff
@@ -26,30 +29,48 @@ class WeekdayActivity : AppCompatActivity() {
         val imageViewCalendarImageWhenEmpty: ImageView = findViewById(R.id.empty_list_image_view)
         val lecturesRecyclerView: RecyclerView = findViewById(R.id.lecture_recycler_view)
         val lecNumberCountTextView: TextView = findViewById(R.id.lecture_number_text_view)
-        val addNewLectureEventFloatingActionButton: FloatingActionButton = findViewById(R.id.lecture_add_floating_action_button)
+        val addNewLectureEventFloatingActionButton: FloatingActionButton =
+            findViewById(R.id.lecture_add_floating_action_button)
+
+
+        val intent = intent
+        val weekDay = intent.getStringExtra("weekday")
+        val weekNum = intent.getStringExtra("weeknum")
+
+        val viewModelFactory = WeekdayActivityViewModelFactory(this, weekDay.toString())
+        val viewModel = ViewModelProvider(this, viewModelFactory).get(WeekdayActvityViewModel::class.java)
+
 
         //recycler View Adapter
-        val timeList: List<String> = listOf(
-            "10:00 - 12:00",
-            "12:00 - 14:00",
-            "14:00 - 16:00",
-            "16:00 - 18:00",
-            "08:00 - 10:00"
-        )
-        val subjectList: List<String> =
-            listOf("Biology", "Math", "Java", "Science", "Python")
+        val timeList: MutableList<String> = mutableListOf()
+        val subjectList: MutableList<String> = mutableListOf()
+
+        viewModel.getLiveLecturesData().observe(this, Observer {
+            if (it.size != 0) {
+                for (i in it.indices) {
+                    timeList.add(i, it.get(i).startTime + "-" + it.get(i).endTime)
+                    subjectList.add(i, it.get(i).lec)
```

Comment:
after you are done adding all the data, you should call adapter.notifyDatasetChanged(), so the recyclerview reflects the changes